testsuite/tests/misc/ephetest2.ml

```diff
@@ -144,9 +144,10 @@ let run test init =
   Stack.clear env.varephe_false;
   Gc.full_major ();
   let res = Hashtbl.fold (fun _ v acc -> acc && check_var v) env.vars true in
-  is_true test "check" res
+  is_true test "check" res;
+  env (* Keep env.varephe_true alive. *)
```

Comment:
Is this change made necessary by this PR, or did it just happen by chance that the problem was not triggered before?

Reply:
I would say that was a latent bug in the test program.